### PR TITLE
Use more readable internal naming scheme.

### PR DESCRIPTION
**Pull request by shurcooL**
The original motivation for using the ugly and non-idiomatic _vfsgen_ prefix for all generated internal types was to ensure no possibility of having a name collision with the user's own code. For example, if user had a type FileInfo, and vfsgen tried to generate FileInfo as well, that'd be a compile time error.

By starting all identifiers with an underscore – something no manually written idiomatic Go code would do – we ensured no collisions.

The idea here is to use a more readable alternative way of achieving the same goal. Since we're using a hard-to-type unicode character, it's unlikely people would write that by hand, so name collisions are still unlikely.

That little dot ۰ is an Arabic zero numeral (U+06F0), categories [Nd].

This idea is inspired by internal code of `golang.org/x/tools/go/ssa/interp` package. See https://github.com/golang/tools/blob/86ad1193da6aba71355e71191641b9cfb26daa67/go/ssa/interp/external.go#L36.

### CODE REVIEW

**Comment by shurcooL** [comment]
@dominikh, would you mind reviewing this proposed change? I am curious what your take on it is.

File: generator.go [hunk 1]
```diff
@@ -412,99 +412,99 @@ func (f *_vfsgen_compressedFile) Read(p []byte) (n int, err error) {
...
+	if d.pos >= len(d.entries) && count > 0 {
 		return nil, io.EOF
 	}
-	if count <= 0 || count > len(d._vfsgen_dirInfo.entries)-d.pos {
-		count = len(d._vfsgen_dirInfo.entries) - d.pos
+	if count <= 0 || count > len(d.entries)-d.pos {
+		count = len(d.entries) - d.pos
 	}
-	e := d._vfsgen_dirInfo.entries[d.pos : d.pos+count]
+	e := d.entries[d.pos : d.pos+count]
```
Comment (dominikh) [inline generator.go]: This is an unrelated change, right?

**Comment by dominikh** [comment]
I am not sure what you're gaining by replacing one scheme with another. Is it to have more readable names in stacktraces? Personally I would be worried about ۰ looking a lot like · and being confusing in output that still uses the latter to separate package name and identifier.

File: generator.go (hunk 1 above)
Comment (shurcooL) [inline generator.go]: Yes, relatively unrelated. I could've done this before too.

I got rid of it because I thought it also increases readability.

**Comment by shurcooL** [comment]
> I am not sure what you're gaining by replacing one scheme with another.

Primarily, it's to make the private/unexported code cleaner and more readable. It's not something other users would look at or need to look at, but I (the developer/maintainer of this library) do. So I want it to be nicer looking and more readable.

> Is it to have more readable names in stacktraces?

I don't expect it to show up in stack traces. Do you mean if there's an internal programming error causing panics? Or is there a valid scenario where vfsgen internal code may show up in a stack trace aside from internal errors?

> Personally I would be worried about ۰ looking a lot like · and being confusing in output that still uses the latter to separate package name and identifier.

The reason I wasn't worried about that is because of point above. Hmm. Not sure if that's a big deal because only one of those characters can exist inside an identifier:

```
main.go:20: invalid identifier character U+00B7 '·'
```

**Comment by dominikh** [comment]
> Do you mean if there's an internal programming error causing panics

Yes.

> Hmm. Not sure if that's a big deal because only one of those characters can exist inside an identifier

I wasn't talking about someone using · in an identifier, but about this:

```
dominikh-pc ~ $ cat /tmp/foo.go
package pkg

func Foo() {}
dominikh-pc ~ $ go build -gcflags=-S /tmp/foo.go |& grep Foo·
"".Foo·f t=9 dupok size=8
```

Though admittedly, the middle dot used to show up in more places than that, but I think Go dropped it in favour of a plain dot in most places, so that's not really much of a concern anymore. For reference, see e.g. commit 3a0fbfab57b133f4703cc7693a2af745570785ba in the Go project to see what I was talking about. Nowadays, it only really shows up when referring to certain runtime symbols .

**Comment by shurcooL** [comment]
> Yes.

I see. That's not a big concern then, because such panics have a low chance of happening. If they do, they can get reported and fixed easily.

I see what you were referring to. It sounds like it's not a big concern by now, and not a blocker for this.

Thanks for taking a look @dominikh!
